ci/test-checks.sh

```diff
@@ -54,8 +54,7 @@ _ cargo +"$rust_stable" fmt --all -- --check
 # -Z... is needed because of clippy bug: https://github.com/rust-lang/rust-clippy/issues/4612
 # run nightly clippy for `sdk/` as there's a moderate amount of nightly-only code there
 _ cargo +"$rust_nightly" clippy \
-  -Zunstable-options --workspace --all-targets \
-  -- --deny=warnings --allow=clippy::stable_sort_primitive
+  -Zunstable-options --workspace --all-targets -- --deny=warnings
```

Comment:
I went ahead and popped `--allow=clippy::stable_sort_primitive` out here, since stable enforces it now